pkg/config/validation/validation.go

```diff
@@ -2044,7 +2050,7 @@ var ValidateVirtualService = registerValidateFunc("ValidateVirtualService",
 			errs = appendErrors(errs, errors.New("http, tcp or tls must be provided in virtual service"))
 		}
 		for _, httpRoute := range virtualService.Http {
-			errs = appendErrors(errs, validateHTTPRoute(httpRoute))
+			errs = appendErrors(errs, validateHTTPRoute(httpRoute, isDelegate))
```

Comment:
You also need to run another check: ensure that the delegation is applied only when the virtual service is bound to gateways and not to the default mesh gateway (i.e. sidecars) - as we have not fully figured out the semantics of the delegation at sidecars.

Reply:
As you said https://github.com/istio/istio/pull/22118#discussion_r405909172

If we require the gateways empty, there is no way to figure out here

Reply:
I meant for the root VS.